app/models/mixins/lifecycle_mixin.rb

```diff
@@ -0,0 +1,21 @@
+module LifecycleMixin
+  extend ActiveSupport::Concern
+
+  def update_lifecycle_state
+    case miq_request.request_state
+    when "finished"
+      lifecycle_state = miq_request.status == 'Ok' ? "provisioned" : "error_provisioned"
+      update(:lifecycle_state => lifecycle_state)
+    else
+      update(:lifecycle_state => 'provisioning')
+    end
+  end
+
+  def provisioned?
+    lifecycle_state == 'provisioned'
```

Comment:
@lfu Would it make sense to have the states as constants and freeze them? (provisioning, provisioned, error_provisioned)